openmeter/customer/repository/customer.go

```diff
@@ -58,13 +69,12 @@ func (r repository) ListCustomers(ctx context.Context, params customer.ListCusto
 	return response, nil
 }
 
+// CreateCustomer creates a new customer
 func (r repository) CreateCustomer(ctx context.Context, params customer.CreateCustomerInput) (*customer.Customer, error) {
-	db := r.client()
-
-	query := db.Customer.Create().
+	// Create the customer in the database
+	query := r.tx.Customer.Create().
```

Comment:
r.client() would be better, as this way you get a panic if this is not executed inside a transaction.

Reply:
It's executed in a TX

Reply:
But that's not explicit, and we will get a panic if something changes. That's why the client is there.

Reply:
Or at least add a nil check for tx if you want to mandate tx.

Reply:
We'll have to rewrite how the transaction is handled to work with any other repositories, so i wouldn't worry too much about it for now